include/tvm/bounded_analyzer.h

```diff
@@ -0,0 +1,43 @@
+#include <tvm/arithmetic.h>
+#include <tvm/ir.h>
+#include <tvm/ir_visitor.h>
+
+namespace tvm {
+namespace ir {
+
+class BoundedAnalyzer final : public IRVisitor {
```

Comment:
How about IRVisitorWithAnalyzer to be consistent with IRMutatorWithAnalyzer, let us move the header into src/arithmetic for now as we want to minimize the amount of public headers(which means we promise stable APIs)